test/jdk/java/awt/color/ICC_Profile/RenderingIntentStressTest.java

```diff
@@ -0,0 +1,97 @@
+import static java.awt.color.ICC_Profile.icICCAbsoluteColorimetric;
+import static java.awt.color.ICC_Profile.icMediaRelativeColorimetric;
+import static java.awt.color.ICC_Profile.icPerceptual;
+import static java.awt.color.ICC_Profile.icRelativeColorimetric;
+import static java.awt.color.ICC_Profile.icSaturation;
+
+/**
+ * @test
+ * @bug 8358057
+ * @summary Stress test for ICC_Profile rendering intent parsing and validation
+ */
+public final class RenderingIntentStressTest {
+
+    public static void main(String[] args) {
+        ICC_Profile builtin = ICC_Profile.getInstance(ColorSpace.CS_sRGB);
+        ICC_Profile profile = ICC_Profile.getInstance(builtin.getData());
+        // some random combinations that should be ignored
+        int[] upperBytes = {0x0000, 0xFFFF, 0xA5A5, 0x8000, 0x0001, 0x8080,
+                            0x0101, 0xAA55, 0x550A, 0xFF00};
+        for (int up : upperBytes) {
+            for (int low = 0; low <= 0xFFFF; low++) {
+                test(profile, up, low);
+            }
+        }
+    }
+
+    private static int getRenderingIntent(byte[] header) {
+        // replicate the logic we have in jdk
+        int index = ICC_Profile.icHdrRenderingIntent;
+        return (header[index + 2] & 0xff) << 8 | header[index + 3] & 0xff;
+    }
+
+    private static void test(ICC_Profile profile, int up, int low) {
+        byte[] header = profile.getData(ICC_Profile.icSigHead);
+        // These bytes should be ignored
+        header[ICC_Profile.icHdrRenderingIntent + 0] = (byte) (up >> 8 & 0xFF);
+        header[ICC_Profile.icHdrRenderingIntent + 1] = (byte) (up & 0xFF);
+        // This is the actual intent
+        header[ICC_Profile.icHdrRenderingIntent + 2] = (byte) (low >> 8 & 0xFF);
+        header[ICC_Profile.icHdrRenderingIntent + 3] = (byte) (low & 0xFF);
+
+        boolean isValid = isValidIntent(low);
+        try {
+            profile.setData(ICC_Profile.icSigHead, header);
+            if (!isValid) {
+                throw new RuntimeException("IAE is expected");
+            }
+        } catch (IllegalArgumentException e) {
+            if (isValid) {
+                throw e;
+            }
+            return;
+        }
+        // verify that the intent is correctly stored in the profile by the CMM
+        byte[] data = profile.getData(ICC_Profile.icSigHead);
+        int actualIntent = getRenderingIntent(data);
+        if (actualIntent != low) {
+            System.out.println("Expected: " + low);
+            System.out.println("Actual: " + actualIntent);
+            throw new RuntimeException("Unexpected intent");
+        }
+    }
+
+    private static boolean isValidIntent(int intent) {
+        return intent == icPerceptual || intent == icRelativeColorimetric
+                || intent == icMediaRelativeColorimetric
+                || intent == icSaturation || intent == icAbsoluteColorimetric
+                || intent == icICCAbsoluteColorimetric;
```

Comment:
I hadn't noticed it earlier but `icAbsoluteColorimetric` and `icICCAbsoluteColorimetric` (since 1.5) point to the same intent. Can it be unified or has it be maintained as two separate constants for backward compatibility ?

Reply:
>I hadn't noticed it earlier but icAbsoluteColorimetric and icICCAbsoluteColorimetric point to the same intent. Is it good to unify them into a single constant in src code during clean up (separate issue)?

At some point, that constants were renamed in the ICC spec, and as a result, the new constants with the different names were added to ICC_Profile. The old one were not removed to avoid breaking compatibility. It wasn't considered critical, since both constants have the same value.

>Since both indicate intent value of 3 only one comparison should be sufficient in the test.

I just included all constants supported by the JDK for completeness.